java/src/main/java/com/powsybl/dataframe/network/extensions/CgmesMetadataModelDataframeProvider.java

```diff
@@ -0,0 +1,70 @@
+
+import com.google.auto.service.AutoService;
+
+import com.powsybl.cgmes.model.CgmesMetadataModel;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.dataframe.network.ExtensionInformation;
+import com.powsybl.dataframe.network.NetworkDataframeMapper;
+import com.powsybl.dataframe.network.NetworkDataframeMapperBuilder;
+import com.powsybl.dataframe.network.adders.NetworkElementAdder;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.cgmes.extensions.CgmesMetadataModels;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * @author Naledi El Cheikh <naledi.elcheikh@rte-france.com>
+ */
+
+@AutoService(NetworkExtensionDataframeProvider.class)
+public class CgmesMetadataModelDataframeProvider extends AbstractSingleDataframeNetworkExtension {
+    @Override
+    public String getExtensionName() {
+        return CgmesMetadataModels.NAME;
+    }
+
+    @Override
+    public ExtensionInformation getExtensionInformation() {
+        return new ExtensionInformation(CgmesMetadataModels.NAME, "Provides information about CGMES metadata models",
+                "index : id (str), cgmes_subset (str), id (str), description (str), " +
+                        "version (int), modeling_authority_set (str), profiles (str), dependent_on (str), supersedes (str) ");
+    }
+
+    private Stream<CgmesMetadataModel> itemsStream(Network network) {
+        if (network.getExtension(CgmesMetadataModels.class) == null) {
+            throw new PowsyblException("No CGMES metadata model found");
+        }
+        return network.getExtension(CgmesMetadataModels.class).getModels().stream();
+    }
+
+    @Override
+    public NetworkDataframeMapper createMapper() {
+        return NetworkDataframeMapperBuilder.ofStream(this::itemsStream)
+                .stringsIndex("id", CgmesMetadataModel::getId)
+                .strings("cgmes_subset", cgmesMetadataModel -> String.valueOf(cgmesMetadataModel.getSubset()))
+                .strings("description", CgmesMetadataModel::getDescription)
+                .ints("version", CgmesMetadataModel::getVersion)
+                .strings("modeling_authority_set", CgmesMetadataModel::getModelingAuthoritySet)
+                .strings("profiles", cgmesMetadataModel -> String.valueOf(cgmesMetadataModel.getProfiles()))
+                .strings("dependent_on", cgmesMetadataModel -> String.valueOf(cgmesMetadataModel.getDependentOn()))
+                .strings("supersedes", cgmesMetadataModel -> String.valueOf(cgmesMetadataModel.getSupersedes()))
+                .build();
+
+    }
+
+    @Override
+    public void removeExtensions(Network network, List<String> ids) {
+        ids.stream().filter(Objects::nonNull)
+                .map(id -> network.getExtension(CgmesMetadataModels.class))
+                .filter(Objects::nonNull)
+                .forEach(model -> ((CgmesMetadataModels) model).getModels().remove(model));
+    }
```

Comment:
It does not work for now (you try to remove multiple times the same extension).
To be consistent with other extensions, this method should remove the individual CgmesMetadataModel by selecting them from the list `ids`